packages/flutter/lib/src/cupertino/sheet_route.dart

```diff
@@ -0,0 +1,237 @@
+        blurRadius: 10,
+        color: CupertinoColors.black.withOpacity(0.12),
+        spreadRadius: 5);
+    final Color backgroundColor = this.backgroundColor ??
+        CupertinoTheme.of(context).scaffoldBackgroundColor;
+    return SafeArea(
+      bottom: false,
+      right: false,
+      left: false,
+      child: Padding(
+        padding: const EdgeInsets.only(top: _kPreviousRouteVisibeOffset),
+        child: Container(
+          clipBehavior: Clip.hardEdge,
+          decoration: BoxDecoration(
+            color: backgroundColor,
+            boxShadow: <BoxShadow>[shadow],
+            borderRadius: BorderRadius.vertical(top: topRadius),
+          ),
+          child: MediaQuery.removePadding(
+            context: context,
+            removeTop: true,
+            child: child,
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+
+
+class CupertinoSheetRoute<T> extends SheetRoute<T> {
+ 
+  CupertinoSheetRoute({
+    required WidgetBuilder builder,
+    double? closeProgressThreshold,
+    String? barrierLabel,
+    Curve? animationCurve,
+    Color? modalBarrierColor,
+    bool bounce = true,
+    bool isDismissible = true,
+    bool enableDrag = true,
+    bool expanded = true,
+    Duration? duration,
+    RouteSettings? settings,
+    this.transitionBackgroundColor,
+    this.topRadius = kCupertinoModalSheetTopRadius,
+    this.previousRouteAnimationCurve,
+  })  : assert(expanded != null),
+        assert(isDismissible != null),
+        assert(enableDrag != null),
+        super(
+          closeProgressThreshold: closeProgressThreshold,
+          builder: (BuildContext context) {
+            return _CupertinoSheetDecorationBuilder(
+              child: Builder(
+                builder: (BuildContext context) {
+                  return builder(context);
+                },
+              ),
+              backgroundColor: transitionBackgroundColor,
+              topRadius: topRadius,
+            );
+          },
+          bounce: bounce,
+          barrierLabel: barrierLabel,
+          modalBarrierColor: modalBarrierColor ?? _kModalBarrierColor,
+          isDismissible: isDismissible,
+          enableDrag: enableDrag,
+          expanded: expanded,
+          settings: settings,
+          animationCurve: animationCurve,
+          duration: duration,
+        );
+
+  /// The top corners of this modal sheet are rounded by this Radius
+  final Radius topRadius;
+
+  /// Background color behind all routes. Black by default
+  final Color? transitionBackgroundColor;
+
+  /// Curve for second animation of previous route transition
+  final Curve? previousRouteAnimationCurve;
+
+
+  @override
+  Widget buildTransitions(
+    BuildContext context,
+    Animation<double> animation,
+    Animation<double> secondaryAnimation,
+    Widget child,
+  ) {
+    final double paddingTop = MediaQuery.of(context)?.padding.top ?? 0;
+    return AnimatedBuilder(
+      animation: secondaryAnimation,
```

Comment:
This is used to animate a CupertinoSheetRoute when the next route is a CupertinoSheetRoute. But like it is now it would always animate when the next route is a SheetRoute, should fix that